src/lib/icli/index.js

```diff
@@ -0,0 +1,357 @@
+'use strict';
+
+const _ = require('lodash');
+
+/**
+ * Private variable to store commander instance
+ */
+let program;
+
+/**
+ * Private variable to store inquirer instance
+ */
+let prompt;
+
+/**
+ * Interactive Command Line Interface
+ * @type {Object}
+ */
+const icli = {
+
+  /**
+   * Colorize a portion of text
+   * @type {Object}
+   */
+  format: {
+    cmd:     msg => { return icli.format.custom(msg, '\x1b[33m'); }, // Yellow
+    info:    msg => { return icli.format.custom(msg, '\x1b[36m'); }, // Cyan
+    error:   msg => { return icli.format.custom(msg, '\x1b[31m'); }, // Red
+    success: msg => { return icli.format.custom(msg, '\x1b[32m'); }, // Green
+    custom: (msg, code) => { return code + msg + '\x1b[0m'; },
+    // aliases
+    ko: msg => { return icli.format.error(msg); },
+    ok: msg => { return icli.format.success(msg); },
+  },
+
+  /**
+   * Commander instance setter
+   * @param {Object} p - commander instance
+   * @return {Object} - the icli instance
+   */
+  setProgram(p) {
+    program = p;
+    return this;
+  },
+
+  /**
+   * Inquirer instance setter
+   * @param {Object} p - inquirer instance
+   * @return {Object} - the icli instance
+   */
+  setPrompt(p) {
+    prompt = p;
+    return this;
+  },
+
+  /**
+   * Create a new interactive command
+   * @param  {Object} config - a configuration object
+   * @param  {function} executeCommand - callback processing the property values once defined by the command line and the prompt
+   * @return {void}
+   */
+  createSubCommand(config, executeCommand) {
+    // create the command
+    const cmd = program.command(config.cmd);
+    cmd.description(config.description);
+
+    // Extract commander arguments and options from the list of parameters
+    // Also enrich parameter configs with a name calculated from "cmdSpec"
+    // ES6 syntax:
+    //   const { args, options } = parseParameters(config.parameters);
+    const tmp = parseParameters(config.parameters);
+    const args = tmp.args;
+    const options = tmp.options;
+
+    // Add command arguments
+    if (args.length) {
+      cmd.arguments(args.join(' '));
+    }
+    // Add command options
+    _.forEach(options, option => {
+      cmd.option(
+        option.flags,
+        option.description,
+        option.coercion
+      );
+    });
+
+    cmd.action(getAction(config.parameters, executeCommand, config.commanderActionHook, config.inquirerPromptHook));
+  },
+
+
+  /**
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Missing JSDoc @returns for function.](https://www.codacy.com/app/lager/lager/file/2779455359/issues/source?bid=3360491&fileBranchId=3365065#l92)